### PR TITLE
Add documentation for updated Fire TV component

### PR DESCRIPTION
**Pull request by JeffLIrion**
**Description:** Add documentation for native support + ADB authentication for Fire TV

**Pull request in [home-assistant](https://github.com/home-assistant/home-assistant) (if applicable):** home-assistant/home-assistant#17767

## Checklist:

- [x] Branch: `next` is for changes and new documentation that will go public with the next [home-assistant](https://github.com/home-assistant/home-assistant) release. Fixes, changes and adjustments for the current release should be created against `current`.
- [ ] The documentation follows the [standards][standards].

[standards]: https://developers.home-assistant.io/docs/documentation_standards.html

### CODE REVIEW

File: source/_components/media_player.firetv.markdown [hunk 1]
```diff
@@ -27,59 +25,56 @@ Steps to configure your Amazon Fire TV stick with Home Assistant:
...
+- **get_source** (*Optional*): Whether or not to retrieve the current app as the source; the default is `true`.
+- **get_sources** (*Optional*): Whether or not to retrieve the running apps as the list of sources; the default is `true`.
+
+### {% linkable_title ADB Authentication (for Fire TV devices with recent software) %}
+
+If you get a "Device authentication required, no keys available" error when trying to setup Fire TV, then you'll need to create an adbkey and add its path to your configuration.  Follow the instructions on this page to connect to your Fire TV from your computer: [Connecting to Fire TV Through adb](https://developer.amazon.com/zh/docs/fire-tv/connecting-adb-to-device.html).  
+
+**Important!**  In the dialog appearing on your Fire TV, you must check the box that says "always allow connections from this device."  ADB authentication in Home Assistant will only work using a trusted key.
+
+Once you've successfully connected to your Fire TV via the command `adb connect <ipaddress>`, the files `adbkey` and `adbkey.pub` will be created on your computer.  Copy these to your Home Assistant folder and add the path to the `adbkey` file to your configuration.
```
Comment (rytilahti) [inline source/_components/media_player.firetv.markdown]: The location of the adb keyfiles should be pointed out. In linux they are stored under `~/.android/` (at least in my case).

File: source/_components/media_player.firetv.markdown [hunk 2]
```diff
@@ -27,59 +25,61 @@ Steps to configure your Amazon Fire TV stick with Home Assistant:
...
-  description: The friendly name of the device.
-  required: false
-  type: string
-  default: Amazon Fire TV
-{% endconfiguration %}
-
-<p class='note warning'>
-Note that python-firetv has support for multiple Amazon Fire TV devices. If you have more than one configured, be sure to specify the device ID in `device`. Run `firetv-server -h` and/or view the source for complete capabilities.
-</p>
+Configuration variables:
```
Comment (frenck) [inline source/_components/media_player.firetv.markdown]: Please use the configuration tags, for more information please see:
https://home-assistant.io/developers/documentation/create_page/#configuration

File: source/_components/media_player.firetv.markdown [hunk 3]
```diff
@@ -27,59 +25,61 @@ Steps to configure your Amazon Fire TV stick with Home Assistant:
...
+- **host** (*Required*): The IP address your Fire TV device.  
+- **name** (*Optional*): The friendly name of the device; the default is 'Amazon Fire TV'.
+- **port** (*Optional*): The port for your Fire TV device; the default is 5555.
+- **adbkey** (*Optional*): The path to your `adbkey` file.  Note that the file `adbkey.pub` must be in the same directory.  
+- **get_source** (*Optional*): Whether or not to retrieve the current app as the source; the default is `true`.
+- **get_sources** (*Optional*): Whether or not to retrieve the running apps as the list of sources; the default is `true`.
+
+### {% linkable_title ADB Authentication (for Fire TV devices with recent software) %}
+
+If you get a "Device authentication required, no keys available" error when trying to setup Fire TV, then you'll need to create an adbkey and add its path to your configuration.  Follow the instructions on this page to connect to your Fire TV from your computer: [Connecting to Fire TV Through adb](https://developer.amazon.com/zh/docs/fire-tv/connecting-adb-to-device.html).
```
Comment (frenck) [inline source/_components/media_player.firetv.markdown]:
```suggestion
If you get a "Device authentication required, no keys available" error when trying to set up Fire TV, then you'll need to create an adbkey and add its path to your configuration.  Follow the instructions on this page to connect to your Fire TV from your computer: [Connecting to Fire TV Through adb](https://developer.amazon.com/zh/docs/fire-tv/connecting-adb-to-device.html).  
```

File: source/_components/media_player.firetv.markdown [hunk 4]
```diff
@@ -27,59 +25,61 @@ Steps to configure your Amazon Fire TV stick with Home Assistant:
...
+Once you've successfully connected to your Fire TV via the command `adb connect <ipaddress>`, the files `adbkey` and `adbkey.pub` will be created on your computer.  The default locations for these files are (from https://developer.android.com/studio/command-line/adb):
+
+* Linux and Mac: `$HOME/.android.`
+* Windows: `%USERPOFILE%\.android.`
+
+Copy the `adbkey` and `adbkey.pub` files to your Home Assistant folder and add the path to the `adbkey` file to your configuration.  
+
+#### ADB Troubleshooting
+
+If you receive the error message `Issue: Error while setting up platform firetv` in your log when trying to setup a Fire TV device with an ADB key, then there is probably an issue with your ADB key.  Here are some possible causes.
```
Comment (frenck) [inline source/_components/media_player.firetv.markdown]:
```suggestion
If you receive the error message `Issue: Error while setting up platform firetv` in your log when trying to set up a Fire TV device with an ADB key, then there is probably an issue with your ADB key.  Here are some possible causes.
```

**Review by frenck — CHANGES_REQUESTED** [review]
@JeffLIrion Thanks for the PR, I've reviewed it and left some comments. 
Could you please take a look? 👍

**Review by frenck — APPROVED** [review]
✅ Approved. Can be merged as soon as the parent PR gets merged.

**Comment by JeffLIrion** [comment]
https://github.com/home-assistant/home-assistant/pull/17767 has been merged!

**Comment by frenck** [comment]
Thanks, @JeffLIrion for letting us know!
Congrats on getting this merged. Great job btw! 🥇

**Pull request merged**